### PR TITLE
fix: refactor the bindgen to use XtpNormalizedType

### LINKED ISSUE #10: Bindgen Refactor
See https://github.com/dylibso/xtp/issues/887

The Go PR maybe be the closest analogue https://github.com/dylibso/xtp-go-bindgen/pull/22

### LINKED ISSUE #11: error when no export function description is in schema
```
Error: failed to render template: Uncaught TypeError: ejs:13
    11| <% if (needsDocumentation(ex)) { %>
    12| /// <summary> <% -%>
 >> 13| <% ex.description.split('\n').filter(line => line.trim() !== "").forEach(line => { %>
    14| /// <%= line.trim() -%>
    15| <% }) %>
    16| /// </summary>

cannot read property 'split' of undefined
```

### PR DESCRIPTION
**Pull request by mhmd-azeez**
Fixes #10 
Fixes #11 

- use `XtpNormalizedType`
- print error messages when there is an exception

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,8 +1,18 @@
 import ejs from "ejs";
-import { getContext, helpers, Import, Export, Property, Schema, XtpSchema } from "@dylibso/xtp-bindgen";
+import { getContext, helpers, Import, Export, Property, Schema, XtpSchema, XtpNormalizedType, ArrayType, BufferType, ObjectType, EnumType, MapType } from "@dylibso/xtp-bindgen";
 
 function toPascalCase(s: string) {
```
Comment (bhelx) [inline src/index.ts]: should we move this to xtp-bindgen? i noticed we do have a `snakeToPascal` helper

File: src/index.ts [hunk 2]
```diff
@@ -13,90 +23,94 @@ function needsDocumentation(x: Export | Import) {
...
-      return "JsonNode";
-    case "array":
-      if (!property.items) return "JsonArray";
-      // TODO this is not quite right to force cast
-      return `${toCSharpType(property.items as Property)}[]`;
+      return "bool";
+    case "date-time":
+      return "DateTime";
+    case "byte":
+      return "byte";
```
Comment (bhelx) [inline src/index.ts]: We're deprecating this so let's remove it. it's unused at the moment

**Review by bhelx — APPROVED** [review]
A few comments, looks good though. thanks!

File: src/index.ts (hunk 1 above)
Comment (mhmd-azeez) [reply src/index.ts]: nice! I am going to use that

**Pull request merged**